course/src/isometry.cc

```diff
@@ -0,0 +1,131 @@
+#include <cmath>
+#include <cstdint>
+#include "isometry.h"
+
+namespace ekumen {
+namespace math {
+
+double& Vector3::operator [] (const int index) const {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
+}
+
+double& Vector3::operator [] (const int index) {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
+}
+
+Vector3 Vector3::operator + (const Vector3& v) const {
+    return Vector3(x_ + v.x_, y_ + v.y_, z_ + v.z_);
+}
+
+Vector3 Vector3::operator - (const Vector3& v) const {
+    return Vector3(x_ - v.x_, y_ - v.y_, z_ - v.z_);
+}
+
+Vector3 Vector3::operator * (const Vector3& v) const {
+    return Vector3(x_ * v.x_, y_ * v.y_, z_ * v.z_);
+}
+
+Vector3 Vector3::operator / (const Vector3&v) const {
+    return Vector3(x_ / v.x_, y_ / v.y_, z_ / v.z_);
+}
+
+Vector3 Vector3::operator = (const Vector3& v) {
```

Comment:
While this will work, it's not efficient. This operator should actually return a `Vector3 &`, and end up with `return *this;`.

I'll let you think what the difference is and share it with the group tomorrow.